api-put-object.go

```diff
@@ -78,16 +78,14 @@ func (opts PutObjectOptions) Header() (header http.Header) {
 	if opts.CacheControl != "" {
 		header["Cache-Control"] = []string{opts.CacheControl}
 	}
-	if opts.EncryptMaterials != nil {
-		header[amzHeaderIV] = []string{opts.EncryptMaterials.GetIV()}
-		header[amzHeaderKey] = []string{opts.EncryptMaterials.GetKey()}
-		header[amzHeaderMatDesc] = []string{opts.EncryptMaterials.GetDesc()}
+	if opts.ServerSide != nil {
+		opts.ServerSide.Marshal(header)
 	}
 	if opts.StorageClass != "" {
 		header[amzStorageClass] = []string{opts.StorageClass}
 	}
 	for k, v := range opts.UserMetadata {
-		if !isAmzHeader(k) && !isStandardHeader(k) && !isSSEHeader(k) && !isStorageClassHeader(k) {
+		if !isAmzHeader(k) && !isStandardHeader(k) && !isStorageClassHeader(k) {
```

Comment:
we should error out if users set conflicting sse-c keys in metadata such that we can ask them to use ServerSide{}

Reply:
Okay - yes, agree.

Reply:
Done - but added this to the `validate()` function - so that the we error out.